subsys/bluetooth/controller/hci/hci.c

```diff
@@ -5100,11 +5104,16 @@ static void le_ext_adv_report(struct pdu_data *pdu_data,
 			sec_phy = sec_phy_curr;
 			data_len = data_len_curr;
 			total_data_len = data_len;
+			total_scan_data_len = 0U;
```

Comment:
Possibly outside the scope of this PR: The do-while loop where this scan data is retrieved is now more than 200 lines of code, inside of a function that is now almost 400 lines of code. Is there any way we can split this into smaller blocks of code without any significant negative changes in performance or memory usage?